parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -306,33 +309,35 @@ private WriteBuilder createContextFunc(
       return this;
     }
 
+    // Utility method to get the column path
+    private String getParquetColumnPath(
```

Comment:
I was wondering, the field Id seems not used, couldnt we just make a map of column name to column path?


```    
 Map<String, String> colNameToPaths =
          parquetSchema.getColumns().stream()
              .filter(col -> col.getPrimitiveType().getId() != null)
              .collect(
                  Collectors.toMap(
                      col ->  {
                             int fieldId = col.getPrimitiveType().getId().intValue();
                             return schema.findColumnName(fieldId);
                        },
                      col -> String.join(".", col.getPath())));
```

Worth to double check the logic, i think this would make the logic simpler if it is correct?

Reply:
would probably need to modify the code to strengthen null handling for the two null cases

Reply:
Thanks @szehon-ho! I will go through your comments and get back tomorrow.

Reply:
@szehon-ho I tried to read this part of code, confused by the column name/column path. I am working with @huaxingao on it. At the same time, what is the difference here? My understanding is that for user config, it will always be the full column path in case of nested fields. Can you explain a bit more? Thanks

Reply:
Talked with @huaxingao, I uploaded a new patch, which uses a map of column name to column path, thanks @huaxingao

Reply:
Thanks ! Sorry for miss the comment, I was out sick

Reply:
Sorry to hear that, hope you fully recovered.